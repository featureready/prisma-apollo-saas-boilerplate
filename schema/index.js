import { gql } from 'apollo-server-express';

import userSchema from './user';
import teamSchema from './team';

const linkSchema = gql`
  scalar DateTime

  directive @requireAuth on FIELD_DEFINITION
  directive @analytics(type: String!, event: String) on FIELD_DEFINITION
  directive @analytics_group(type: String!, event: String) on FIELD_DEFINITION
  directive @computed(value: String) on FIELD_DEFINITION

  type Query {
    _: Boolean
  }
  type Mutation {
    _: Boolean
  }
  # type Subscription {
  #   _: Boolean
  # }
`;

export default [linkSchema, userSchema, teamSchema];
