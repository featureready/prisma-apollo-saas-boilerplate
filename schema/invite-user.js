import { gql } from 'apollo-server-express';

export default gql`
  extend type Mutation {
    InviteUser(input: InviteUserInput!): Invite! @requireAuth
  }

  input InviteUserInput {
    firstName: String!
    lastName: String!
    email: String!
  }

  type Invite {
    id: String!
    email: String!
    firstName: String!
    lastName: String!
    status: String!
  }
`;
