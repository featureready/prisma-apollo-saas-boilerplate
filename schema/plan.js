import { gql } from 'apollo-server-express';

export default gql`
  extend type Query {
    Plans: [Plan] @requireAuth
  }

  type Plan {
    id: String!
    title: String
    description: String
    amount: Int!
    currency: String!
    interval: String!
  }
`;
