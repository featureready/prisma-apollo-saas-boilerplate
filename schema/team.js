import { gql } from 'apollo-server-express';

export default gql`
  extend type Mutation {
    createTeam(name: String!): Team!
      @requireAuth
      @analytics(type: "track", event: "Create Team")
      @analytics_group(type: "group")
    updateTeam(name: String!): Team! @requireAuth @analytics(type: "track", event: "Update Team")
  }

  type Team {
    id: String
    name: String
  }
`;
