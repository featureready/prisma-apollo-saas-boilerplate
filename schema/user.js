import { gql } from 'apollo-server-express';

export default gql`
  extend type Query {
    me: User @requireAuth
    paymentHistory: [Invoice] @requireAuth
  }

  extend type Mutation {
    signup(firstName: String!, lastName: String!, email: String!, password: String!): AuthPayload!
      @analytics(type: "identify")
    login(email: String!, password: String!): AuthPayload! @analytics(type: "identify")
    forgotPassword(email: String!): Result
    resetPassword(password: String!, token: String!): Result
    updateUser(firstName: String, lastName: String, email: String, password: String): User!
      @requireAuth
    deleteUser(id: ID!): Int!
    addCreditCard(token: String): Result @requireAuth
    subscribePlan(planId: String!): Result @requireAuth
  }

  type AuthPayload {
    jwt: String!
    user: User!
  }

  type Result {
    message: String
  }

  type User {
    id: String
    firstName: String
    lastName: String
    fullName: String @computed(value: "$firstName $lastName")
    email: String
    createdAt: DateTime
    updatedAt: DateTime
    periodStart: DateTime
    periodEnd: DateTime
    team: Team
  }

  type Invoice {
    amountDue: Int
    amountPaid: Int
    invoicePdf: String
    status: String
    date: DateTime
    periodStart: DateTime
    periodEnd: DateTime
  }
`;
