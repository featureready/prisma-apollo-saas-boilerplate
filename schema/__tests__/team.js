import { mockServer } from 'graphql-tools';
import schema from '../index';

const mocks = {
  Boolean: () => false,
  ID: () => '1',
  Int: () => 1,
  Float: () => 12.34,
  String: () => 'String',
  DateTime: () => new Date(2018, 8, 8)
};

const server = mockServer(schema, mocks, false);

describe('Team schema', () => {
  it('has valid type definitions', () => {
    expect(() => server.query(`{ __schema { types { name } } }`)).not.toThrow();
  });

  it(`mutation { createTeam } should return expected value`, () => {
    const payload = `
      mutation CreateTeam($name: String!) {
        createTeam(name: $name) {
          id
          name
        }
      }
    `;

    const variables = {
      name: 'Awesome team'
    };

    expect(server.query(payload, variables)).resolves.toMatchSnapshot();
  });

  it(`mutation { updateTeam } should return expected value`, () => {
    const payload = `
      mutation UpdateTeam($name: String!) {
        updateTeam(name: $name) {
          id
          name
        }
      }
    `;

    const variables = {
      name: 'Awesome team 2'
    };

    expect(server.query(payload, variables)).resolves.toMatchSnapshot();
  });
});
