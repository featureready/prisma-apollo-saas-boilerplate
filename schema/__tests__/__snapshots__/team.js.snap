// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team schema mutation { createTeam } should return expected value 1`] = `
Object {
  "data": Object {
    "createTeam": Object {
      "id": "String",
      "name": "String",
    },
  },
}
`;

exports[`Team schema mutation { updateTeam } should return expected value 1`] = `
Object {
  "data": Object {
    "updateTeam": Object {
      "id": "String",
      "name": "String",
    },
  },
}
`;
